src/promptflow-core/promptflow/executor/_script_executor.py

```diff
@@ -72,6 +72,15 @@ def __init__(
         self._message_format = MessageFormatType.BASIC
         self._multimedia_processor = BasicMultimediaProcessor()
 
+    def _get_func_name(self):
+        try:
+            original_func = getattr(self._func, "__original_function")
```

Comment:
async function?

Reply:
I guess we can store the name once we initialize this executor.

Reply:
For async function, seems `self._func` also store a sync function? 
![image](https://github.com/microsoft/promptflow/assets/7776147/23c2f47d-f28e-47c2-9290-93b2e624e1f3)

Reply:
stored to executor's field